Dockerfile

```diff
@@ -23,3 +23,14 @@ COPY --from=builder /usr/bin/k6 /usr/bin/k6
 USER 12345
 WORKDIR /home/k6
 ENTRYPOINT ["k6"]
+
+# Legacy loadimpact/k6 image
+FROM grafana/k6
+
+# Temporarily switch back to root to move files
+USER 0
+
+RUN mv /usr/bin/k6 /usr/bin/k6.original
```

Comment:
That line concerns me since we don't copy just built binary, but moving the one that was in `grafana/k6`  :thinking:

Reply:
That's the point, the idea is to replace `k6` binary with an entrypoint script, that displays the warning and then runs the original binary.

Reply:
Oh, I think I've missed your point. Right, I'm using `grafana/k6` as a base for this image to avoid re-building the binary for `loadimpact/k6`, but since we're building and tagging a new version `grafana/k6` in previous step, this is the one that will be used as a base.

Reply:
Yeah, but I'm not sure if it's actually using the local built image (https://github.com/grafana/k6/actions/runs/5268112485/jobs/9524337784?pr=3121): 

```
#9 [linux/arm64 stage-2 1/3] FROM docker.io/grafana/k6@sha256:acad1a06a5e32b5f04f9dbfad112075c8c71b6be8d664bd9d8f79ce6b9559b3d
#9 sha2[56](https://github.com/grafana/k6/actions/runs/5268112485/jobs/9524337784?pr=3121#step:11:57):166b2807661b8e8712a96c60d16bb08ec088eb6a725c6746252a5017a176329f 12.56MB / 12.56MB 0.2s done
#9 sha256:211f8214916f225d429bb667937f6c1bed8d1693e6c081158347b940e4f0f55c 286.20kB / 286.20kB 0.2s done
#9 extracting sha256:211f8214916f225d429bb667937f6c1bed8d1693e6c0811[58](https://github.com/grafana/k6/actions/runs/5268112485/jobs/9524337784?pr=3121#step:11:59)347b940e4f0f55c 0.1s done
#9 extracting sha256:166b2807661b8e8712a96c60d16bb08ec088eb6a725c6746252a5017a176329f
#9 extracting sha256:166b2807661b8e8712a96c[60](https://github.com/grafana/k6/actions/runs/5268112485/jobs/9524337784?pr=3121#step:11:61)d16bb08ec088eb6a725c6746252a5017a176329f 0.3s done
#9 DONE 0.5s
```

Reply:
We can avoid the user change, and the copying of the binary if we simply override the entrypoint for this stage. I.e.:

```dockerfile
# Legacy loadimpact/k6 image
FROM --platform=$BUILDPLATFORM release

COPY entrypoint-legacy.sh /usr/bin/
ENTRYPOINT ["/usr/bin/entrypoint-legacy.sh"]
```

Then the entrypoint script can just call the `k6` binary directly.